src/component.jl

```diff
@@ -269,19 +269,20 @@ function get_time_series_by_key(
     )
 end
 
-function assign_new_uuid!(component::InfrastructureSystemsComponent)
+function assign_new_uuid_internal!(component::InfrastructureSystemsComponent)
     old_uuid = get_uuid(component)
     new_uuid = make_uuid()
     if has_time_series(component)
         container = get_time_series_container(component)
         # There may be duplicates because of transform operations.
         changed_uuids = Set{Tuple{Base.UUID, String}}()
         for (key, ts_metadata) in container.data
-            changed_uuid = (old_uuid, key.name)
+            ts_uuid = get_time_series_uuid(ts_metadata)
+            changed_uuid = (ts_uuid, key.name)
```

Comment:
This is an existing bug that affects some use case in PowerSystems, like `convert_component!`. Users would know if they experienced it. We could back-port the if necessary. Since no one has reported it, that is probably not needed.

Reply:
ok